### ISSUE TITLE
Restore `dbt.` prefix for all migrated cross-db macros

### ISSUE THREAD

**Issue by joellabes** [dbt-labs/dbt-utils#690]
### Describe the bug
In https://github.com/dbt-labs/dbt-core/issues/5888, we discovered that dbt projects that already contained a macro with the same name (e.g. `concat()`) as a cross-db macro were taking precedence in an unexpected manner. 

We want to switch back to explicitly stating the namespace (i.e. `dbt.concat()`) so that the dispatch logic works correctly (https://github.com/dbt-labs/dbt-core/issues/5720 and https://github.com/dbt-labs/dbt-core/pull/5907).

### Additional context
The current best way I've come up with to handle this is a regex find and replace, using the [list of cross-db macros from the docs site](https://docs.getdbt.com/reference/dbt-jinja-functions/cross-database-macros):
```
(any_value|bool_or|cast_bool_to_text|concat|dateadd|datediff|date_trunc|escape_single_quotes|except|hash|intersect|last_day|length|listagg|position|replace|right|safe_cast|split_part|string_literal|type_bigint|type_float|type_int|type_numeric|type_string|type_timestamp|type_bigint|type_float|type_int|type_numeric|type_string|type_timestamp|except|intersect|concat|hash|length|position|replace|right|split_part|escape_single_quotes|string_literal|any_value|bool_or|listagg|cast_bool_to_text|safe_cast|dateadd|datediff|date_trunc|last_day)
```
Replacing `{1}` with `dbt.{1}` ([I only know one trick, but it's a good trick](https://discourse.getdbt.com/t/migrating-away-from-dbt-utils-not-null-where-and-unique-where-tests/2773))

@jtcohen6: given that https://github.com/dbt-labs/dbt-core/pull/5907 is v1.3 only, should this change be made available to both dbt-utils 0.9.x (requires dbt Core 1.2+) and dbt-utils 1.0 (requires dbt Core 1.3+) or only the latter? Do you have a read on which approach will be safer?

### Are you interested in contributing the fix?
I'd love to see a community contribution here - happy to help out if the specifics wind up being gnarly.

**Comment by jtcohen6** [comment]
> @jtcohen6: given that https://github.com/dbt-labs/dbt-core/pull/5907 is v1.3 only, should this change be made available to both dbt-utils 0.9.x (requires dbt Core 1.2+) and dbt-utils 1.0 (requires dbt Core 1.3+) or only the latter? Do you have a read on which approach will be safer?

I don't have a perfect read here. The bug was quite niche. It only cropped up if you have a custom generic test that calls a macro which _used_ to live in `dbt_utils` and has since moved to `dbt`. Some of the generic test

For now, it's probably slightly safer to just make this change for dbt-utils v1.0. We could consider backporting the fix from https://github.com/dbt-labs/dbt-core/pull/5907 for a dbt-core v1.2.x patch release, if we don't see any negative repercussions. It would make this decision simpler.

**Comment by joellabes** [comment]
> For now, it's probably slightly safer to just make this change for dbt-utils v1.0

Agreed! 🔨🧑‍⚖️

**Comment by joellabes** [comment]
Resolved by #701
